Compilation warnings on Linux

**Description of the problem**

```
gmtmex.c: In function ‘mexFunction’:
gmtmex.c:213:3: warning: this ‘if’ clause does not guard... [-Wmisleading-indentation]
   if (!pPersistent || (API = (void *)pPersistent[0]) == NULL)
   ^~
gmtmex.c:215:4: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘if’
    mexAtExit(force_Destroy_Session); /* Register an exit function. */
    ^~~~~~~~~
gcc -I/home/runner/work/gmtmex/gmt-install-dir/include/gmt -I/usr/include/octave-4.2.2/octave -O2 -Wall -m64 -fPIC -fno-strict-aliasing -std=c99 -DGMT_OCTMEX -I/home/runner/work/gmtmex/gmt-install-dir/include/gmt -c gmtmex_parser.c
gmtmex_parser.c: In function ‘gmtmex_palette_init’:
gmtmex_parser.c:1106:44: warning: variable ‘minmax’ set but not used [-Wunused-but-set-variable]
   double *colormap = NULL, *range = NULL, *minmax = NULL, *alpha = NULL, *bfn = NULL, *hinge = NULL, *cpt = NULL, *cyclic = NULL;
                                            ^~~~~~
gmtmex_parser.c:1106:11: warning: variable ‘colormap’ set but not used [-Wunused-but-set-variable]
   double *colormap = NULL, *range = NULL, *minmax = NULL, *alpha = NULL, *bfn = NULL, *hinge = NULL, *cpt = NULL, *cyclic = NULL;
           ^~~~~~~~
```

--- a/.github/workflows/ci-tests.yml
+++ b/.github/workflows/ci-tests.yml
@@ -20,7 +20,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [ubuntu-20.04, macos-latest]
+        os: [ubuntu-latest, macos-latest]
         gmt_version: [6.1.1, master]
 
     env:
@@ -41,6 +41,7 @@ jobs:
           if [ "$RUNNER_OS" == "macOS" ]; then
             brew install octave
           elif [ "$RUNNER_OS" == "Linux" ]; then
+            sudo apt-get update
             sudo apt-get install octave liboctave-dev
           fi
 

--- a/.github/workflows/ci-tests.yml
+++ b/.github/workflows/ci-tests.yml
@@ -20,7 +20,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [ubuntu-latest, macos-latest]
+        os: [ubuntu-20.04, macos-latest]
         gmt_version: [6.1.1, master]
 
     env:

--- a/src/gmtmex.c
+++ b/src/gmtmex.c
@@ -91,7 +91,7 @@ static void *Initiate_Session (unsigned int verbose) {
 
 static void *alloc_default_plhs (void *API, struct GMT_RESOURCE *X) {
 	/* Allocate a default plhs when it was not stated in command line. That is, mimic the Matlab behavior
-	   when we do for example (i.e. no lhs):  sqrt([4 9])  
+	   when we do for example (i.e. no lhs):  sqrt([4 9])
 	*/
 	void *ptr = NULL;
 	switch (X->family) {
@@ -143,14 +143,14 @@ void mexFunction (int nlhs, mxArray *plhs[], int nrhs, const mxArray *prhs[]) {
 		char message[128] = {""};
 		sprintf (message, "Warning: Your GMT version (%d.%d.%d) may be too new to work with GMTMEX %d.%d.%d.\n",
 			GMT_MAJOR_VERSION, GMT_MINOR_VERSION, GMT_RELEASE_VERSION, GMTMEX_GMT_MAJOR_VERSION, GMTMEX_GMT_MINOR_VERSION, GMTMEX_GMT_PATCH_VERSION);
-		mexPrintf (message); 
+		mexPrintf (message);
 	}
 	else if (GMT_MAJOR_VERSION < GMTMEX_GMT_MAJOR_VERSION || GMT_MINOR_VERSION < GMTMEX_GMT_MINOR_VERSION || (GMT_MINOR_VERSION == GMTMEX_GMT_MINOR_VERSION && GMT_RELEASE_VERSION < GMTMEX_GMT_PATCH_VERSION)) {
 		char message[128] = {""};
 		sprintf (message, "Error: The GMT shared library must be at least version %d.%d.%d but you have %d.%d.%d.\n",
 			GMTMEX_GMT_MAJOR_VERSION, GMTMEX_GMT_MINOR_VERSION, GMTMEX_GMT_PATCH_VERSION,
 			GMT_MAJOR_VERSION, GMT_MINOR_VERSION, GMT_RELEASE_VERSION);
-		mexErrMsgTxt (message); 
+		mexErrMsgTxt (message);
 	}
 
 	/* 0. No arguments at all results in the GMT banner message */
@@ -160,7 +160,7 @@ void mexFunction (int nlhs, mxArray *plhs[], int nrhs, const mxArray *prhs[]) {
 	}
 
 	/* 1. Check for the special commands create and help */
-	
+
 	if (nrhs == 1) {	/* This may be create or help */
 		cmd = mxArrayToString (prhs[0]);
 		if (!cmd) mexErrMsgTxt("GMT: First input argument must be a string. Maybe a composition of a string and a cell array?\n");
@@ -201,7 +201,7 @@ void mexFunction (int nlhs, mxArray *plhs[], int nrhs, const mxArray *prhs[]) {
 		}
 		else
 			API = (void *)pPersistent[0];       /* Get the GMT API pointer */
-		if (API == NULL) mexErrMsgTxt ("GMT: This GMT5 session has is corrupted. Better to start from scratch.\n"); 
+		if (API == NULL) mexErrMsgTxt ("GMT: This GMT5 session has is corrupted. Better to start from scratch.\n");
 	}
 	else if (mxIsScalar_(prhs[0]) && mxIsUint64(prhs[0])) {
 		/* Here, nrhs > 1 . If first arg is a scalar int, we assume it is the API memory address */
@@ -210,9 +210,10 @@ void mexFunction (int nlhs, mxArray *plhs[], int nrhs, const mxArray *prhs[]) {
 		first = 1;		/* Commandline args start at prhs[1] since prhs[0] had the API id argument */
 	}
 	else {		/* We still don't have the API, so we must get it from the past or initiate a new session */
-		if (!pPersistent || (API = (void *)pPersistent[0]) == NULL)
+		if (!pPersistent || (API = (void *)pPersistent[0]) == NULL) {
 			API = Initiate_Session (verbose);	/* Initializing new GMT session */
 			mexAtExit(force_Destroy_Session);	/* Register an exit function. */
+		}
 #endif
 	}
 
@@ -239,14 +240,14 @@ void mexFunction (int nlhs, mxArray *plhs[], int nrhs, const mxArray *prhs[]) {
 	}
 
 	/* 2. Get module name and separate out args */
-	
+
 	/* Here we have a GMT module call. The documented use is to give the module name separately from
 	 * the module options, but users may forget and combine the two.  So we check both cases. */
-	
+
 	n_in_objects = nrhs - 1;
 	str_length = strlen (cmd);				/* Length of module (or command) argument */
 	for (k = 0; k < str_length && cmd[k] != ' '; k++);	/* Determine first space in command */
-	
+
 	if (k == str_length) {	/* Case 2a): No spaces found: User gave 'module' separately from 'options' */
 		strcpy (module, cmd);				/* Isolate the module name in this string */
 		if (nrhs > 1 && mxIsChar (prhs[first+1])) {	/* Got option string */
@@ -297,11 +298,11 @@ void mexFunction (int nlhs, mxArray *plhs[], int nrhs, const mxArray *prhs[]) {
 
 	/* Make sure this is a valid module */
 	if ((status = GMT_Call_Module (API, module, GMT_MODULE_EXIST, NULL)) != GMT_NOERROR) 	/* No, not found */
-		mexErrMsgTxt ("GMT: No module by that name was found.\n"); 
-	
+		mexErrMsgTxt ("GMT: No module by that name was found.\n");
+
 	/* Below here we may actually wish to add options to the opt_args, but it is a pointer.  So we duplicate to
 	 * another string with enough space. */
-	
+
 	if (opt_args) strcpy (opt_buffer, opt_args);	/* opt_buffer has lots of space for additions */
 	/* 2+ Add -F to psconvert if user requested a return image but did not explicitly give -F */
 	if (!strncmp (module, "psconvert", 9U) && nlhs == 1 && (!opt_args || !strstr ("-F", opt_args))) {	/* OK, add -F */
@@ -310,7 +311,7 @@ void mexFunction (int nlhs, mxArray *plhs[], int nrhs, const mxArray *prhs[]) {
 		else
 			strcpy (opt_buffer, "-F");
 	}
-	
+
 	/* 2++ If gmtwrite then add -T? with correct object type */
 	if (strstr(module, "write") && opt_args && !strstr(opt_args, "-T") && n_in_objects == 1) {	/* Add type for writing to disk */
 		char targ[5] = {" -T?"};
@@ -327,7 +328,7 @@ void mexFunction (int nlhs, mxArray *plhs[], int nrhs, const mxArray *prhs[]) {
 
 	if (!options && nlhs == 0 && nrhs == 1 && strcmp (module, "end")) 	/* Just requesting usage message, so add -? to options */
 		options = GMT_Create_Options (API, 0, "-?");
-	
+
 	/* 4. Preprocess to update GMT option lists and return info array X */
 	if ((X = GMT_Encode_Options (API, module, n_in_objects, &options, &n_items)) == NULL) {
 		if (n_items == UINT_MAX)	/* Just got usage/synopsis option */
@@ -335,15 +336,15 @@ void mexFunction (int nlhs, mxArray *plhs[], int nrhs, const mxArray *prhs[]) {
 		else
 			mexErrMsgTxt ("GMT: Failure to encode mex command options\n");
 	}
-	
+
 	if (options) {	/* Only for debugging - remove this section when stable */
 		gtxt = GMT_Create_Cmd (API, options);
 		GMT_Report (API, GMT_MSG_DEBUG, "GMT_Encode_Options: Revised command after memory-substitution: %s\n", gtxt);
 		GMT_Destroy_Cmd (API, &gtxt);	/* Only needed it for the above verbose */
 	}
-	
+
 	/* 5. Assign input sources (from MATLAB to GMT) and output destinations (from GMT to MATLAB) */
-	
+
 	for (k = 0; k < n_items; k++) {	/* Number of GMT containers involved in this module call */
 		if (X[k].direction == GMT_IN) {
 			if ((X[k].pos+first+1) < (unsigned int)nrhs)
@@ -361,7 +362,7 @@ void mexFunction (int nlhs, mxArray *plhs[], int nrhs, const mxArray *prhs[]) {
 		}
 		GMTMEX_Set_Object (API, &X[k], ptr);	/* Set object pointer */
 	}
-	
+
 	/* 6. Run GMT module; give usage message if errors arise during parsing */
 	status = GMT_Call_Module (API, module, GMT_MODULE_OPT, options);
 	if (status != GMT_NOERROR) {
@@ -374,7 +375,7 @@ void mexFunction (int nlhs, mxArray *plhs[], int nrhs, const mxArray *prhs[]) {
 	}
 
 	/* 7. Hook up any GMT outputs to MATLAB plhs array */
-	
+
 	for (k = 0; k < n_items; k++) {	/* Get results from GMT into MATLAB arrays */
 		if (X[k].direction == GMT_IN) continue;	/* Only looking for stuff coming OUT of GMT here */
 		pos = X[k].pos;		/* Short-hand for index into the plhs[] array being returned to MATLAB */
@@ -386,7 +387,7 @@ void mexFunction (int nlhs, mxArray *plhs[], int nrhs, const mxArray *prhs[]) {
 		GMT_Set_Default (API, "API_PAD", "2");
 
 	/* 8. Free all GMT containers involved in this module call */
-	
+
 	for (k = 0; k < n_items; k++) {
 		void *ppp = X[k].object;
 		if (GMT_Close_VirtualFile (API, X[k].name) != GMT_NOERROR)
@@ -400,7 +401,7 @@ void mexFunction (int nlhs, mxArray *plhs[], int nrhs, const mxArray *prhs[]) {
 	}
 
 	/* 9. Destroy linked option list */
-	
+
 	if (GMT_Destroy_Options (API, &options) != GMT_NOERROR)
 		mexErrMsgTxt ("GMT: Failure to destroy GMT5 options\n");
 #ifdef SINGLE_SESSION

--- a/src/gmtmex_parser.c
+++ b/src/gmtmex_parser.c
@@ -222,7 +222,7 @@ static void *gmtmex_get_dataset (void *API, struct GMT_DATASET *D) {
 		D_struct = mxCreateStructMatrix (0, 0, N_MEX_FIELDNAMES_DATASET, GMTMEX_fieldname_dataset);
 		return (D_struct);
 	}
-	
+
 	for (tbl = seg_out = 0; tbl < D->n_tables; tbl++)	/* Count non-zero segments */
 		for (seg = 0; seg < D->table[tbl]->n_segments; seg++)
 			if (D->table[tbl]->segment[seg]->n_rows)
@@ -280,7 +280,7 @@ static void *gmtmex_get_postscript (void *API, struct GMT_POSTSCRIPT *P) {
 	uint64_t k, *length = NULL;
 	unsigned int *mode = NULL;
 	mxArray *P_struct = NULL, *mxptr[N_MEX_FIELDNAMES_PS], *mxstring = NULL;
-	
+
 	if (P == NULL)	/* Safety valve */
 		mexErrMsgTxt ("gmtmex_get_postscript: programming error, input POSTSCRIPT struct P is NULL or data string is empty\n");
 
@@ -289,7 +289,7 @@ static void *gmtmex_get_postscript (void *API, struct GMT_POSTSCRIPT *P) {
 		P_struct = mxCreateStructMatrix (0, 0, N_MEX_FIELDNAMES_PS, GMTMEX_fieldname_ps);
 		return P_struct;
 	}
-	
+
 	/* Return PS with postscript and length in a struct */
 	P_struct = mxCreateStructMatrix (1, 1, N_MEX_FIELDNAMES_PS, GMTMEX_fieldname_ps);
 
@@ -299,15 +299,15 @@ static void *gmtmex_get_postscript (void *API, struct GMT_POSTSCRIPT *P) {
 	mxptr[3] = mxCreateCellMatrix (P->n_headers, P->n_headers ? 1 : 0);
 	length   = (uint64_t *)mxGetData(mxptr[1]);
 	mode     = (uint32_t *)mxGetData(mxptr[2]);
-	
+
 	length[0] = (uint64_t)P->n_bytes;	/* Set length of the PS string */
 	mode[0]   = (uint32_t)P->mode;		/* Set mode of the PS string */
-	
+
 	for (k = 0; k < P->n_headers; k++) {
 		mxstring = mxCreateString (P->header[k]);
 		mxSetCell (mxptr[3], (int)k, mxstring);
 	}
-	
+
 	for (k = 0; k < N_MEX_FIELDNAMES_PS; k++)
 		mxSetField (P_struct, 0, GMTMEX_fieldname_ps[k], mxptr[k]);
 
@@ -344,7 +344,7 @@ static void *gmtmex_get_palette (void *API, struct GMT_PALETTE *C) {
 		C_struct = mxCreateStructMatrix (0, 0, N_MEX_FIELDNAMES_CPT, GMTMEX_fieldname_cpt);
 		return (C_struct);
 	}
-	
+
 	/* Return CPT via colormap, range, and alpha arrays in a struct */
 	/* Create a MATLAB struct for this CPT */
 	C_struct = mxCreateStructMatrix (1, 1, N_MEX_FIELDNAMES_CPT, GMTMEX_fieldname_cpt);
@@ -361,7 +361,7 @@ static void *gmtmex_get_palette (void *API, struct GMT_PALETTE *C) {
 	mxptr[8] = NULL;	/* Set below */
 	mxptr[9] = mxCreateNumericMatrix (1, 1, mxDOUBLE_CLASS, mxREAL);
 	mxptr[10] = mxCreateCellMatrix (C->n_headers, C->n_headers ? 1 : 0);
-	
+
 	color    = mxGetPr (mxptr[0]);
 	alpha    = mxGetPr (mxptr[1]);
 	range    = mxGetPr (mxptr[2]);
@@ -429,7 +429,7 @@ static void *gmtmex_get_image (void *API, struct GMT_IMAGE *I) {
 	mxptr[3]  = mxCreateNumericMatrix (1, 6, mxDOUBLE_CLASS, mxREAL);
 	mxptr[4]  = mxCreateNumericMatrix (1, 2, mxDOUBLE_CLASS, mxREAL);
 	mxptr[5]  = mxCreateDoubleScalar ((double)I->header->registration);
-	mxptr[6]  = mxCreateDoubleScalar ((double)I->header->nan_value);	
+	mxptr[6]  = mxCreateDoubleScalar ((double)I->header->nan_value);
 	mxptr[7]  = mxCreateString (I->header->title);
 	mxptr[8]  = mxCreateString (I->header->remark);
 	mxptr[9]  = mxCreateString (I->header->command);
@@ -459,7 +459,7 @@ static void *gmtmex_get_image (void *API, struct GMT_IMAGE *I) {
 			color[k] = (uint8_t)I->colormap[k];
 		k /= 4;
 		memcpy (u, I->data, I->header->nm * sizeof (uint8_t));
-	}	
+	}
 	else if (I->header->n_bands == 1) {	/* gray image */
 		mxptr[0] = mxCreateNumericMatrix (I->header->n_rows, I->header->n_columns, mxUINT8_CLASS, mxREAL);
 		u = mxGetData (mxptr[0]);
@@ -481,7 +481,7 @@ static void *gmtmex_get_image (void *API, struct GMT_IMAGE *I) {
 		if (I->alpha) {
 			mxptr[15] = mxCreateNumericMatrix (I->header->n_rows, I->header->n_columns, mxUINT8_CLASS, mxREAL);
 			alpha = mxGetData (mxptr[15]);
-			memcpy (alpha, I->alpha, I->header->nm * sizeof (uint8_t)); 
+			memcpy (alpha, I->alpha, I->header->nm * sizeof (uint8_t));
 		}
 	}
 	else if (I->header->n_bands == 4) {	/* RGBA image, with a color map */
@@ -490,8 +490,8 @@ static void *gmtmex_get_image (void *API, struct GMT_IMAGE *I) {
 		u = mxGetData (mxptr[0]);
 		mxptr[15] = mxCreateNumericMatrix (I->header->n_rows, I->header->n_columns, mxUINT8_CLASS, mxREAL);
 		alpha = mxGetData (mxptr[15]);
-		memcpy (u, I->data, 3 * I->header->nm * sizeof (uint8_t)); 
-		memcpy (alpha, &(I->data)[3 * I->header->nm], I->header->nm * sizeof (uint8_t)); 
+		memcpy (u, I->data, 3 * I->header->nm * sizeof (uint8_t));
+		memcpy (alpha, &(I->data)[3 * I->header->nm], I->header->nm * sizeof (uint8_t));
 		/*
 		for (k = 0; k < I->header->nm; k++) {
 			for (m = 0; m < 3; m++)
@@ -609,7 +609,7 @@ static struct GMT_GRID *gmtmex_grid_init (void *API, unsigned int direction, uns
 				free (str);
 			}
 			mx_ptr = mxGetField (ptr, 0, "wkt");
-			if (mx_ptr != NULL && mxGetN(mx_ptr) > 20) {	/* A true WTT string will have more thna this lenght */ 
+			if (mx_ptr != NULL && mxGetN(mx_ptr) > 20) {	/* A true WTT string will have more thna this lenght */
 				char *str = malloc(mxGetN(mx_ptr) + 1);
 				mxGetString(mx_ptr, str, (mwSize)mxGetN(mx_ptr) + 1);
 				G->header->ProjRefWKT = GMT_Duplicate_String (API, str);
@@ -803,7 +803,7 @@ static struct GMT_IMAGE *gmtmex_image_init (void *API, unsigned int direction, u
 			free (str);
 		}
 		mx_ptr = mxGetField (ptr, 0, "wkt");
-		if (mx_ptr != NULL && mxGetN(mx_ptr) > 20) {	/* A true WTT string will have more than this length */ 
+		if (mx_ptr != NULL && mxGetN(mx_ptr) > 20) {	/* A true WTT string will have more than this length */
 			char *str = malloc(mxGetN(mx_ptr) + 1);
 			mxGetString(mx_ptr, str, (mwSize)mxGetN(mx_ptr) + 1);
 			I->header->ProjRefWKT = GMT_Duplicate_String (API, str);
@@ -937,7 +937,7 @@ static void *gmtmex_dataset_init (void *API, unsigned int direction, unsigned in
 		/* 1. A dataset MATLAB structure or array of structures.
 		 * 2. A Cell array of plain text strings for a text-only file.
 		 * 3. A single text string instead of a one-item cell array of strings. */
-		
+
 		if (mxIsStruct (ptr)) {	/* Got the dataset structure */
 			dim[GMT_SEG] = mxGetM (ptr);	/* Number of segments */
 			if (dim[GMT_SEG] == 0) mexErrMsgTxt ("gmtmex_dataset_init: Input has zero segments where it can't be.\n");
@@ -1113,7 +1113,7 @@ static struct GMT_PALETTE *gmtmex_palette_init (void *API, unsigned int directio
 			if ((mx_ptr[k] = mxGetField (ptr, 0, GMTMEX_fieldname_cpt[k])) == NULL)
 				gmtmex_quit_if_missing ("gmtmex_palette_init", GMTMEX_fieldname_cpt[k]);
 		}
-		
+
 		dim[0] = mxGetM (mx_ptr[0]);	/* Number of rows in colormap */
 		if (dim[0] < 1)
 			mexErrMsgTxt ("gmtmex_palette_init: Colormap array has no CPT values\n");
@@ -1126,6 +1126,10 @@ static struct GMT_PALETTE *gmtmex_palette_init (void *API, unsigned int directio
 		hinge    = mxGetData (mx_ptr[6]);
 		cpt      = mxGetData (mx_ptr[7]);
 		cyclic   = mxGetData (mx_ptr[9]);
+
+		/* Disable unused-but-set-variable warnings */
+		(void)(minmax);
+		(void)(colormap);
 
 		dim[1] = mxGetM (mx_ptr[2]);	/* Length of range array */
 		if (dim[0] > dim[1]) {	/* This only happens when we have a continuous color table */
@@ -1210,7 +1214,7 @@ static struct GMT_POSTSCRIPT *gmtmex_ps_init (void *API, unsigned int direction,
 			if ((mx_ptr[k] = mxGetField (ptr, 0, GMTMEX_fieldname_ps[k])) == NULL)
 				gmtmex_quit_if_missing ("gmtmex_ps_init", GMTMEX_fieldname_ps[k]);
 		}
-			
+
 		length = mxGetData (mx_ptr[1]);
 		if (length[0] == 0)
 			mexErrMsgTxt ("gmtmex_ps_init: Dimension of PostScript given as zero\n");

